pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java

```diff
@@ -211,6 +211,10 @@ private static void validateFieldConfigList(TableConfig tableConfig) {
   private static void validateValidationConfig(TableConfig tableConfig) {
     SegmentsValidationAndRetentionConfig validationConfig = tableConfig.getValidationConfig();
     if (validationConfig != null) {
+      String timeColumnName = validationConfig.getTimeColumnName();
+      if(timeColumnName == null) {
+        throw new IllegalStateException("Must provide valid timeColumnName in tableConfig");
+      }
```

Comment:
Only force time column for real-time table, and please fix the format
```suggestion
      if (tableConfig.getTableType() == TableType.REALTIME && validationConfig.getTimeColumnName() == null) {
        throw new IllegalStateException("Must provide time column in real-time table config");
      }
```